core/block/editor/userdataobject/userdata.go

```diff
@@ -0,0 +1,271 @@
+package userdataobject
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	anystore "github.com/anyproto/any-store"
+	"github.com/anyproto/any-store/anyenc"
+	"github.com/anyproto/any-sync/util/crypto"
+	"golang.org/x/exp/slices"
+
+	"github.com/anyproto/anytype-heart/core/block/cache"
+	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
+	"github.com/anyproto/anytype-heart/core/block/editor/storestate"
+	"github.com/anyproto/anytype-heart/core/block/source"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+)
+
+var log = logging.Logger("core.block.editor.userdata")
+
+type UserDataObject interface {
+	smartblock.SmartBlock
+
+	SaveContact(ctx context.Context, profile *model.IdentityProfile, key crypto.SymKey) error
+	DeleteContact(ctx context.Context, identity string) error
+	UpdateContactByDetails(ctx context.Context, id string, details *domain.Details) error
+	UpdateContactByIdentity(ctx context.Context, profile *model.IdentityProfile) (err error)
+	ListContacts(ctx context.Context) ([]*Contact, error)
+}
+
+type userDataObject struct {
+	smartblock.SmartBlock
+	state       *storestate.StoreState
+	storeSource source.Store
+	crdtDb      anystore.DB
+	arenaPool   *anyenc.ArenaPool
+
+	objectCache      cache.ObjectGetter
+	ctx              context.Context
+	cancel           context.CancelFunc
+	onUpdateCallback func()
+}
+
+func New(sb smartblock.SmartBlock, crdtDb anystore.DB, objectCache cache.ObjectGetter) UserDataObject {
```

Comment:
Do we use the same crdt db as for the account object?

Reply:
Yes, because they both in tech space. Is it a problem?